internal/controller/compute/statefulserverset/statefulserverset.go

```diff
@@ -368,69 +374,67 @@ func isALanFieldNotUpToDate(specLan v1alpha1.StatefulServerSetLan, gotLan v1alph
 	return false
 }
 
-func areDataVolumesUpToDate(cr *v1alpha1.StatefulServerSet, volumes []v1alpha1.Volume) (bool, bool) {
+func areDataVolumesUpToDateAndAvailable(cr *v1alpha1.StatefulServerSet, volumes []v1alpha1.Volume) (creationVolumesUpToDate, areVolumesUpToDate, volumesAvailable bool) {
 	crExpectedNrOfVolumes := len(cr.Spec.ForProvider.Volumes) * cr.Spec.ForProvider.Replicas
 	if len(volumes) != crExpectedNrOfVolumes {
-		return false, false
+		return false, false, false
 	}
 	for volumeIndex := range volumes {
 		for _, specVolume := range cr.Spec.ForProvider.Volumes {
 			if isAVolumeFieldNotUpToDate(specVolume, volumeIndex, volumes) {
-				return true, false
+				return true, false, false
+			}
+			if volumes[volumeIndex].Status.AtProvider.State != ionoscloud.Available {
+				return true, true, false
 			}
 		}
 	}
-	return true, true
+	return true, true, true
 }
 
 func isAVolumeFieldNotUpToDate(specVolume v1alpha1.StatefulServerSetVolume, volumeIndex int, volumes []v1alpha1.Volume) bool {
-	if generateProviderNameFromIndex(specVolume.Metadata.Name, volumeIndex) != volumes[volumeIndex].Spec.ForProvider.Name {
```

Comment:
Why was this removed? Was it because it did not match the way we create the name of the volume (e.g. `<nameFromSpec-replicaIdx-volumeIdx>`)
![image](https://github.com/ionos-cloud/crossplane-provider-ionoscloud/assets/73175496/35b7b8f6-c90d-465f-bd56-f26670c84759)
Shouldn't we in this case just update it to what it should be instead of removing it?

Reply:
changing metadata name means a new object is created. We could support metadata name update, but it gets tricky. We should actually just make them all immutable for now.